Sources/ArrayExtensions.swift

```diff
@@ -201,6 +203,16 @@ extension Array where Element: Equatable {
     }
 }
 
+extension Array where Element: Hashable {
+
+    /// EZSE: Removes all occurrences of the given object(s)
+    public mutating func removeAll(_ elements: [Element]) {
+        let elementsSet = Set(elements)
```

Comment:
Exactly, the `filter` applied upon `self` is `O(n)` (for the `Hashable` as well as for the `Equatable` overload), but given `Hashable` elements, we can avoid the nested predicate in the `filter` closure (namely the `contains` call) from also being "`O(n)`" (here w.r.t. scaling size of the input argument `elements`; I guess we should call it `O(m)` :) ).

Could be a corner-case optimization, but for similarly scaling size of `self` and `elements` (when supplied as an array), `Hashable` elements would have `O(n)` scaling for the call (`O(n)` filter and `O(1)` `contains` predicate) whereas non-`Hashable` elements would be `O(n^2)` (`O(n)` filter and `O(n)` `contains` predicate).